### DISCUSSION TITLE
How do I add styles to child elements?

### DISCUSSION THREAD

**Discussion by therealsujitk** [daohoangson/flutter_widget_from_html#652] [Q&A]
I'm trying to add a style to all the children of a certain element. I noticed the `Element` class has a `styles` member. Is there a way I can add a style map or something to it?

```dart
HtmlWidget(
  _htmlBody,
  customStylesBuilder: (el) {
    if (el.classes.contains('class-name')) {
      var children = el.getElementsByTagName('div')[0].children;

      for (int i = 0; i < children.length; ++i) {
        var child = children[i];
        // Add style to child here
      }
    }

    return null;
  }
),
```

**Comment by daohoangson** [comment]
It's possible to iterate through the element children, you will need to handle the recursive loop yourself though.

Another way to do this is using a custom `WidgetFactory` and register a `BuildOp`. The op has a `onChild` callback which will run for each child.

**Reply by therealsujitk** [reply]
I have iterated through the children in the example above, just not sure what to do next. Can you give me some example(s)?
